words/tools.asm

```diff
@@ -99,121 +99,108 @@ z_dot_s:        rts
         ; DUMP's exact output is defined as "implementation dependent".
         ; This is in assembler because it is
         ; useful for testing and development, so we want to have it work
-        ; as soon as possible. Uses TMP2
+        ; as soon as possible. Uses tmp1, tmp2
         ; """
 
 xt_dump:
                 jsr underflow_2
 w_dump:
 _row:
-                ; start counter for 16 numbers per row
-                ldy #16
-
-                ; We use TMP2 as the index for the ASCII characters
-                ; that we print at the and of the hex block. We
-                ; start saving them at HERE (CP)
```

Comment:
as i was fooling with the unsafe `s"` the use of `(cp)` here was breaking tests, so I figured it was safer to have it run in two passes rather than buffering.  for user-facing, the speed difference won't be significant